openquakeplatform/openquakeplatform/static/js/hrde_hazard_chart_mixed.js

```diff
@@ -0,0 +1,439 @@
+
+        var str = selectedCurves[i];
+        str = str.replace(/_/g, " ");
+        var curveTitle = capitalize(str)
+
+        makeCircles(data, i, color, curveTitle);
+
+        legend.append("text")
+            .attr("x", 90)
+            .attr("y", 20*(i+1))
+            .attr("dy", ".35em")
+            .text(curveTitle);
+
+        legend.append("svg:circle")
+            .attr("cy", 20*(i+1))
+            .attr("cx", 80)
+            .attr("r", 3)
+            .style("fill", color)
+
+        $("."+selectedCurves[i]).css({'stroke':colors[i]});
+    }
+
+    for (i = 0 ; i < xAxis_n ; i++) {
+        var g = svg.append("g");
+
+        g.attr("class", "x axis")
+
+        g.attr("transform", "translate(0," + height + ")")
+        .call(xAxis[i]);
+        if (i == (xAxis_n - 1))
+            g.append("text")
+            .attr("x", 160)
+            .attr("y", 30)
+            .attr("dy", ".71em")
+            .attr("text-anchor", "middle")
+            .style("font-size","12px")
+            .text(xAxisLable);
+    }
+    svg.append("g")
+        .attr("class", "y axis")
+        .call(yAxis)
+        .append("text")
+        .attr("transform", "rotate(-90)")
+        .attr("y", -60)
+        .attr("x", -20)
+        .attr("dy", ".71em")
+        .style("font-size","12px")
+        .style("text-anchor", "end")
+        .text(yAxisLable);
+
+    textTopTitle = svg.append("text")
+        .attr("x", 0)
+        .attr("y", -47)
+        .attr("dy", ".35em")
+        .style("font-weight", "bold")
+        .attr("font-size","14px")
+        .text(AppVars.mappedValue);
+
+    textTopSubTitle = svg.append("text")
+        .attr("x", 0)
+        .attr("y", -32)
+        .attr("dy", ".35em")
+        .attr("font-size","12px")
+        .text(chartHeader);
+
+    textTopSubTitle = svg.append("text")
+        .attr("x", 0)
+        .attr("y", -20)
+        .attr("dy", ".35em")
+        .attr("font-size","12px")
+        .text("(Lon/Lat): "+lon+", "+lat);
+
+    textBottom = svg.append("text")
+        .attr("x", 0)
+        .attr("y", 315)
+        .attr("dy", ".35em")
+        .style("font-size","11px")
+        .text('');
+
+    $('#chartDialog').append('<div id="downloadCurve"><font color="blue">Download Curve</font></div>');
+    $('#downloadCurve').on("hover", function(){
+        $(this).css("cursor", "pointer");
+    });
+
+    var h = $("#chartDialog").height();
+    h = h + 40;
+    $("#chartDialog").css({"height": h+"px"});
+
+    // Prep data for download to CSV
+    $('#downloadCurve').click(function(event) {
+        var csvHeader = selectedCurves;
+        var csvData = [];
+        csvData = csvData.concat(csvHeader);
+        csvData.push("investigationTime");
+        csvData.push("poE");
+        csvData.push("lon");
+        csvData.push("lat");
+        csvData = JSON.stringify(csvData);
+        var lineBreak = "lineBreak";
+        csvData = csvData.concat(lineBreak);
```

Comment:
```concat``` method concatenates the instance array with arrays passed as arguments.
At this line and many others below the ```push``` method is more correct and efficient to just add a new element at the end of array (without reassign it).